apps/src/templates/courseOverview/AssignCourse.js

```diff
@@ -0,0 +1,189 @@
+ * lets them assign the current course to any of those sections (after accepting
+ * a confirmation dialog)
+ */
+class AssignCourse extends Component {
+  static propTypes = {
+    courseId: PropTypes.number.isRequired,
+    courseName: PropTypes.string.isRequired,
+    sectionsInfo: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      name: PropTypes.string.isRequired,
+    })).isRequired,
+  };
+
+  state = {
+    dropdownOpen: false,
+    sectionIndexToAssign: null,
+    errorString: null
+  }
+
+  componentDidMount = () => {
+    document.addEventListener('click', this.onClickDocument, false);
+  }
+
+  componentWillUnmount = () => {
+    document.removeEventListener('click', this.onClickDocument, false);
+  }
+
+  onClickDocument = event => {
+    // We're only concerned with clicks outside of ourselves
+    if (ReactDOM.findDOMNode(this).contains(event.target)) {
+      return;
+    }
+    if (this.state.dropdownOpen) {
+      this.setState({
+        dropdownOpen: false
+      });
+    }
+  }
+
+  onClickDropdown = () => {
+    this.setState({
+      dropdownOpen: !this.state.dropdownOpen
+    });
+  }
+
+  onClickCourse = event => {
+    const sectionIndex = event.target.getAttribute('data-section-index');
+    this.setState({
+      sectionIndexToAssign: sectionIndex
+    });
+  }
+
+  onCloseDialog = event => {
+    this.setState({sectionIndexToAssign: null});
+  }
+
+  updateCourse = () => {
+    const section = this.props.sectionsInfo[this.state.sectionIndexToAssign];
+    $.ajax({
+      url: `/sections/${section.id}`,
+      method: 'PATCH',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify({
+        course_id: this.props.courseId
+      }),
+    }).done(result => {
+      this.setState({dropdownOpen: false});
+    }).fail((jqXhr, status) => {
+      this.setState({
+        dropdownOpen: false,
+        sectionIndexToAssign: null,
+        errorString: i18n.unexpectedError()
+      });
+    });
+  }
+
+  acknowledgeError = () => {
+    this.setState({errorString: null});
+  }
+
+  render() {
+    const { courseId, courseName, sectionsInfo } = this.props;
+    const { dropdownOpen, sectionIndexToAssign, errorString } = this.state;
+    const section = sectionsInfo[sectionIndexToAssign];
+
+    return (
+      <div>
+        <ProgressButton
+          ref={element => this.button = element}
+          text={i18n.assignCourse()}
+          style={styles.button}
+          onClick={this.onClickDropdown}
+          icon={dropdownOpen ? "caret-up" : "caret-down"}
+          iconStyle={styles.icon}
+          color={ProgressButton.ButtonColor.orange}
+        />
+
+        {dropdownOpen && (
+          <div style={styles.dropdown}>
+            <a
+              href={`${window.dashboard.CODE_ORG_URL}/teacher-dashboard?newSection=${courseId}#/sections`}
+              style={styles.section}
+            >
+              {i18n.newSection()}...
```

Comment:
The ellipsis should be part of the localized string, especially for future RTL support.